app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -449,6 +449,16 @@ protected void onResume() {
             sharedPreferences.edit().putBoolean(Constants.KEY_MAIN_PAGE_CHANGE, false).apply();
             NavigationHelper.openMainActivity(this);
         }
+
+        if (sharedPreferences.getBoolean(Constants.KEY_ENABLE_WATCH_HISTORY, true)) {
```

Comment:
This should not introduce a new field in the Constants file, instead you should get the key with `getString(R.string.enable_watch_history_key)`.

It comes from this file, where all the setting key are stored (this is done to avoid bugs so we don't need to hard code every time):

https://github.com/TeamNewPipe/NewPipe/blob/0447e4e6645504d24a7752e2f266aa8b2364d4b0/app/src/main/res/values/settings_keys.xml#L174